temporalcli/commands.server.go

```diff
@@ -0,0 +1,81 @@
+package temporalcli
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/temporalio/cli/temporalcli/devserver"
+)
+
+func (t *TemporalServerStartDevCommand) run(cctx *CommandContext, args []string) error {
+	// Prepare options
+	opts := devserver.StartOptions{
+		FrontendIP:       t.Ip,
+		FrontendPort:     t.Port,
+		Namespaces:       append([]string{"default"}, t.Namespace...),
+		Logger:           cctx.Logger,
+		DatabaseFile:     t.DbFilename,
+		MetricsPort:      t.MetricsPort,
+		FrontendHTTPPort: t.HttpPort,
+	}
+	// Setup UI
+	if !t.Headless {
+		opts.UIIP, opts.UIPort = t.Ip, t.UiPort
+		if opts.UIIP == "" {
+			opts.UIIP = t.Ip
+		}
+		if opts.UIPort == 0 {
+			opts.UIPort = t.Port + 1000
+		}
+		opts.UIAssetPath, opts.UICodecEndpoint = t.UiAssetPath, t.UiCodecEndpoint
+	}
+	// Pragmas and dyn config
+	var err error
+	if opts.SqlitePragmas, err = stringKeysValues(t.SqlitePragma); err != nil {
+		return fmt.Errorf("invalid pragma: %w", err)
+	} else if opts.DynamicConfigValues, err = stringKeysJSONValues(t.DynamicConfigValue); err != nil {
+		return fmt.Errorf("invalid dynamic config values: %w", err)
+	}
+	// If not using DB file, set persistent cluster ID
+	if t.DbFilename == "" {
+		opts.ClusterID = persistentClusterID()
+	}
+	// Log config if requested
+	if t.LogConfig {
+		opts.LogConfig = func(b []byte) {
+			cctx.Logger.Info("Logging config")
+			_, _ = cctx.Options.Stderr.Write(b)
```

Comment:
IIRC it wasn't formatted well since it's YAML. Also, this is an option that is asking for output. I would totally support putting this in stdout though so people can pipe it to a file or something.